logictest: rework default config sets

RaduBerinde opened the pull request:
This change reworks the default config sets into a map, which makes
them easier to read and allows adding new config sets without
refactoring various pieces of code.

The only functional difference is that if a test specifies
`default-configs` explicitly, that now refers to the global defaults,
and not the defaults that may have been overridden for that test
(which can be achieved by not specifying anything).

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/135743)
<!-- Reviewable:end -->

RaduBerinde committed:
logictest: rework default config sets

This change reworks the default config sets into a map, which makes
them easier to read and allows adding new config sets without
refactoring various pieces of code.

The only functional difference is that if a test specifies
`default-configs` explicitly, that now refers to the global defaults,
and not the defaults that may have been overridden for that test
(which can be achieved by not specifying anything).

Epic: none
Release note: None

rafiss reviewed (approved):
lgtm! seems clearer this way

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135743#-:-OC9Szyi3Tkn6cfhQTBF:b2hpsxe)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!

bors r+
